go.mod

```diff
@@ -17,12 +17,12 @@ require (
 	github.com/apparentlymart/go-userdirs v0.0.0-20200915174352-b0c018a67c13
 	github.com/apparentlymart/go-versions v1.0.2
 	github.com/armon/circbuf v0.0.0-20190214190532-5111143e8da2
-	github.com/aws/aws-sdk-go-v2 v1.23.2
-	github.com/aws/aws-sdk-go-v2/feature/ec2/imds v1.14.6
-	github.com/aws/aws-sdk-go-v2/service/dynamodb v1.25.5
-	github.com/aws/aws-sdk-go-v2/service/kms v1.26.5
-	github.com/aws/aws-sdk-go-v2/service/s3 v1.46.0
-	github.com/aws/smithy-go v1.17.0
+	github.com/aws/aws-sdk-go-v2 v1.32.7
```

Comment:
`note`
The new conditional writes feature was added in s3/1.69.0.
Due to a [breaking change](https://github.com/aws/aws-sdk-go-v2/issues/2370#issuecomment-1953308268) that was added in github.com/aws/aws-sdk-go-v2@1.24.0, all deps under the `aws-sdk-go-v2` namespace needs to be updated.

Since the s3 version we need is from 2024.11.25, All the other versions are the versions released right before the s3 one. Most of the versions are from 2024.11.18.